app/controllers/users_controller.rb

```diff
@@ -0,0 +1,60 @@
+class UsersController < ApplicationController
+  before_action :authorized, only: [:show]
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.create(user_params)
+    if @user.valid?
+      @user.save
+      redirect_to @user
+    else
+      redirect_to root_path
+    end
+  end
+
+  def show
+    @user = User.find(params[:id])
```

Comment:
This line is used in like 5 different places in this file. I would rip it out of those 5 places and replace it with a helper method to remove duplicated code. Something like:

```
def user
    @user ||= User.find(params[:id])
end
helper_method :user
```

And then, you will just be able to call `user` instead of needing to call `@user`.